src/mainloop.rkt

```diff
@@ -41,15 +42,14 @@
   (when (not (equal? newval 'none)) (set-shellstate-patched! (^shell-state^) newval))
   (shellstate-patched (^shell-state^)))
 
-;; Iteration 0 alts (original alt in every repr, constant alts, etc.)
-(define (starting-alts altn ctx)
-  (filter alt-expr
-    (for/list ([(k v) (in-hash (*conversions*))]
-              #:unless (equal? k (context-repr ctx))
-              #:when (set-member? v (context-repr ctx)))
-      (define rewrite (get-rewrite-operator k))
-      (define body* (apply-repr-change-expr (list rewrite (alt-expr altn)) ctx))
-      (make-alt body*))))
+;; Iteration 0 expressions (original in every repr, constant alts, etc.)
+(define (starting-expressions expression context)
+  (for/list ([(k v) (in-hash (*conversions*))]
+             #:unless (equal? k (context-repr context))
+             #:when (set-member? v (context-repr context)))
+    (apply-repr-change-expr
+     (list (get-rewrite-operator k) expression)
+     context)))
```

Comment:
Is there a worry that this returns false? I think that's what the `filter` was doing. (You could wrap it in `filter identity`?)

Reply:
I think I might need a little help understanding what this does so I can add it back properly. Right now I've just excluded it and left a TODO.